src/trace/dma-trace.c

```diff
@@ -159,27 +159,31 @@ int dma_trace_host_buffer(struct dma_trace_data *d,
 static int dma_trace_buffer_init(struct dma_trace_data *d)
 {
 	struct dma_trace_buf *buffer = &d->dmatb;
+	void *buf;
+	unsigned int flags;
 
 	/* allocate new buffer */
-	buffer->addr = rballoc(RZONE_BUFFER,
-			       SOF_MEM_CAPS_RAM | SOF_MEM_CAPS_DMA,
-			       DMA_TRACE_LOCAL_SIZE);
-	if (!buffer->addr) {
-		trace_buffer_error("dma_trace_buffer_init() error: "
-				   "alloc failed");
```

Comment:
@mmaka1 This trace error message might still be received by host via mailbox route, so I think we should keep it here.

Reply:
I had assumed this specific -ENOMEM error code is sufficient to find the cause of failure. However others may rely on the error message read from the log, so reverted it.